packages/react-liveness/src/components/FaceLivenessDetector/service/utils/StreamRecorder/StreamRecorder.ts

```diff
@@ -123,6 +133,55 @@ export class StreamRecorder {
     });
   }
 
+  /**
+   * The startFaceLivenessSession API takes in a ReadableStream as its source of all websocket events
+   * To allow for changing cameras we add new MediaRecorders which pass through events to the previously set MediaRecorder
+   */
+  #createPassThroughRecorder(): { [key: string]: (args: any) => void } {
+    const onDataAvailableHandler = ({ data }: { data: Blob }) => {
+      this.#initialRecorder.dispatchEvent(
+        new MessageEvent('dataavailable', { data })
+      );
+    };
+    this.#recorder.ondataavailable = onDataAvailableHandler;
+
+    const onSessionInfoHandler = (e: any) => {
+      const { data } = e as unknown as StreamResult<'sessionInfo'>;
+      this.#initialRecorder.dispatchEvent(
+        new MessageEvent('sessionInfo', { data })
+      );
+    };
+    this.#recorder.addEventListener('sessionInfo', onSessionInfoHandler);
+
+    const onStreamStopHandler = () => {
+      this.#initialRecorder.dispatchEvent(new MessageEvent('streamStop'));
+    };
+    this.#recorder.addEventListener('streamStop', onStreamStopHandler);
+
+    const onCloseCodeHandler = (e: Event) => {
+      const { data } = e as unknown as StreamResult<'closeCode'>;
+      this.#initialRecorder.dispatchEvent(
+        new MessageEvent('closeCode', { data })
+      );
+    };
+    this.#recorder.addEventListener('closeCode', onCloseCodeHandler);
+
+    const onEndStreamHandler = () => {
+      this.#initialRecorder.dispatchEvent(new MessageEvent('endStream'));
+    };
+    this.#recorder.addEventListener('endStream', onEndStreamHandler);
+
+    this.#setupCallbacks();
+
+    return {
+      endStream: onEndStreamHandler,
+      closeCode: onCloseCodeHandler,
+      streamStop: onStreamStopHandler,
+      sessionInfo: onSessionInfoHandler,
+      dataavailable: onDataAvailableHandler,
```

Comment:
Can just set the handlers on `this.#eventHandlers` instead of returning the handlers?

Reply:
haha yes, i'm not sure why i didn't do that